### PR TITLE
Ajax integration

### PR DESCRIPTION
**Pull request by Iamyuvraj143**
<kpd>add to cart1</kpd>

<kpd>add_to_cart2 </kpd>

### CODE REVIEW

File: app/views/products/show.html.erb [hunk 1]
```diff
@@ -23,10 +23,17 @@
...
     </div>
   </div></br>
+</div
 <div class= "container">
   <% if @product.photos.attached? %>
     <% @product.photos.each do |photo| %>
       </br><%= image_tag photo, width: 1000, height: 700 %></br>
     <% end %>
   <% end %>
-</div>
```
Comment (Iamyuvraj143) [inline app/views/products/show.html.erb:32]: ma'am I am not getting, why it is showing this "no new line at the end of file" here

File: app/controllers/cart_products_controller.rb [hunk 2]
```diff
@@ -1,19 +1,26 @@
 class CartProductsController < ApplicationController
   before_action :authenticate_user!
   before_action :load_cart, only: %i( new create edit update destroy)
-  before_action :check_product_stock, only: %i(new)
+  before_action :load_cart_for_update, only: %i(update)
 
   def new
+    @product = Product.find(@product_id)
```
Comment (kalpana19sharma) [inline app/controllers/cart_products_controller.rb]: Use find_by

File: app/controllers/cart_products_controller.rb [hunk 3]
```diff
@@ -27,16 +34,22 @@ def destroy
...
   end
 
-  def load_cart
-    @cart = current_user.shopping_cart
-    @product_id = params[:product_id]
-    @product = Product.find_by(id:[@product_id])
+  def load_cart_for_update
+    quantity = params[:qty].to_i
+    cart_product = @cart.cart_products.find_by(id: params[:id])
+    if cart_product.present? && cart_product.product.quantity>=quantity
```
Comment (kalpana19sharma) [inline app/controllers/cart_products_controller.rb]:
```suggestion
    if cart_product.present? && cart_product.product.quantity >= quantity
```

File: app/controllers/cart_products_controller.rb [hunk 4]
```diff
@@ -27,16 +34,22 @@ def destroy
...
+      cart_product.update(quantity:quantity)
```
Comment (kalpana19sharma) [inline app/controllers/cart_products_controller.rb]: Update logic should be in update action

File: app/controllers/orders_controller.rb [hunk 5]
```diff
@@ -137,12 +137,14 @@ def load_order_address
 
   def place_batch_order
     @order_list.each do |item|
-      @order = current_user.orders.new(address: @address, status:"In Process")
-      @order.save
-      @order_item = @order.order_items.create(product_id:item.product_id, quantity:item.quantity, price:item.total)
-      item.product.quantity -= item.quantity.to_i
-      item.product.save
-      @cart_products.delete(item) 
+      if (item.product.quantity>=item.quantity)
```
Comment (kalpana19sharma) [inline app/controllers/orders_controller.rb]:
```suggestion
      if (item.product.quantity >= item.quantity)
```

File: app/views/addresses/new.js.erb [hunk 6]
```diff
@@ -0,0 +1 @@
+$("#addNewAddressForm").html("<%= escape_javascript render('form', address: @user.address)%>");
```
Comment (kalpana19sharma) [inline app/views/addresses/new.js.erb:1]:
```suggestion
$("#addNewAddressForm").html("<%= j render('form', address: @user.address)%>");
```

File: app/views/cart_products/new.js.erb [hunk 7]
```diff
@@ -0,0 +1 @@
+$("#addToCartForm").html("<%=  escape_javascript render('form', cartProduct: @cart_product)%>");
```
Comment (kalpana19sharma) [inline app/views/cart_products/new.js.erb]:
```suggestion
$("#addToCartForm").html("<%= j render('form', cartProduct: @cart_product)%>");
```

File: app/views/shopping_cart/index.html.erb [hunk 8]
```diff
@@ -41,3 +43,20 @@
...
+    data: { qty: qty},
+    dataType: "json",
+
+    success: function(result){
+        alert(result.d);
+        console.log(result);
+    }
+  });
+  }
+</script>
```
Comment (kalpana19sharma) [inline app/views/shopping_cart/index.html.erb]: this logic should be in JavaScript file

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

**Review by kalpana19sharma — APPROVED** [review]

**Pull request merged**